src/ORM/DataList.php

```diff
@@ -1162,7 +1162,7 @@ private function fetchEagerLoadHasOne(
                     }
                 }
             } else {
-                throw new LogicException("Couldn't find parent for record $fetchedID on $relationType relation $relationName");
+                throw new LogicException("Couldn't find parent for record {$fetched->ID} on $relationType relation $relationName");
```

Comment:
This is unrelated - I noticed while I was looking at this that the `$fetchedID` variable doesn't exist.